modepy/spaces.py

```diff
@@ -48,26 +59,24 @@
 class FunctionSpace(ABC):
     r"""An opaque object representing a finite-dimensional function space
     of functions :math:`\mathbb{R}^n \to \mathbb{R}`.
-
-    .. attribute:: spatial_dim
-
-        :math:`n` in the above definition, the number of spatial dimensions
-        in which the functions in the space operate.
-
-    .. attribute:: space_dim
-
-        The number of dimensions of the function space.
     """
 
+    @property
+    def order(self) -> int:
+        """Polynomial degree of the function space, if any."""
+        raise AttributeError(f"{type(self).__name__} has no attribute 'order'")
```

Comment:
I think we discussed that it's not a good idea to add this "order" thing to the base `FunctionSpace` class. I added it here to silence mypy, which started complaining due to some additional return value typing.

I can remove it if you still think it's a good idea.

Reply:
I'm OK with keeping it. I like the 'if any', to emphasize that this is not always a well-defined notion.